addons/link_tracker/models/link_tracker.py

```diff
@@ -112,26 +113,60 @@ def _get_title_from_url(self, url):
 
         return title
 
-    @api.model
-    def create(self, vals):
-        create_vals = vals.copy()
+    @api.constrains('url', 'campaign_id', 'medium_id', 'source_id')
+    def _check_unicity(self):
+        """
+        Can not be implemented in SQL because we want to care about null values.
+        """
+        # build a query to fetch all needed link trackers at once
+        search_query = []
+        for tracker in self:
+            search_query = ['|'] + search_query + [
+                '&', '&', '&',
+                ('url', '=', tracker.url),
+                ('campaign_id', '=', tracker.campaign_id.id),
+                ('medium_id', '=', tracker.medium_id.id),
+                ('source_id', '=', tracker.source_id.id)
+            ]
+        all_link_trackers = self.search(search_query[1:])
+
+        # check for unicity
+        for tracker in self:
+            if all_link_trackers.filtered(
+                lambda l: l.url == tracker.url
+                and l.campaign_id == tracker.campaign_id
+                and l.medium_id == tracker.medium_id
+                and l.source_id == tracker.source_id
+            ) != tracker:
+                raise UserError(_('Link Tracker values must be unique'))
 
-        if 'url' not in create_vals:
+    @api.model
+    def search_or_create(self, vals):
+        if 'url' not in vals:
             raise ValueError('URL field required')
```

Comment:
Note that we could translate message :) (and maybe improve it: "Creating a Link Tracker without URL is not possible" )